From b4bf53573118529d62c2b32a37aa7d1b44bb8492 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Tue, 23 Apr 2013 14:13:09 -0500
Subject: [PATCH 233/254] arm: socfpga: Fix error for FPGA LED PIO base
 address

Upstream: git://git.rocketboards.org/linux-socfpga.git

The base address for the LED PIO is incorrect. The current address is for
the LWH2F bridge and 0xff2100040 is the address for the LED PIO slave.
The current implementation works because Qsys selected the LED PIO as the
default slave, so all reads/writes from the LWH2F bridge master targeting
holes in the master address map gets routed to the LED PIO anywa.

However, if any peripheral with an address range larger than the PIO's gets
added to the system, then the build malfunctions, since that peripheral
becomes the new default slave.

Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
(cherry picked from commit acb6c4e6fdc4efee1de3afc0cf614485245ea69f)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/socfpga.dtsi |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index 7432420..4923289 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -69,7 +69,7 @@
 		agpio0: gpio@0xc0000000 {
 			compatible = "altr,pio-1.0";
 			/* Register base 0xff200000 is for a light-weight bridge */
-			reg = <0xff200000 0x10>;
+			reg = <0xff210040 0x10>;
 			width = <32>;
 			/* There are 64 interrupts from the FPGA start at 72, so 45 has to be wrong */
 			interrupts = <0 45 4>;
-- 
1.7.5.4

